Source/WebCore/Modules/push-api/PushDatabase.cpp

```diff
@@ -624,21 +639,34 @@ void PushDatabase::getIdentifiers(CompletionHandler<void(HashSet<PushSubscriptio
     });
 }
 
-void PushDatabase::getTopics(CompletionHandler<void(Vector<String>&&)>&& completionHandler)
+void PushDatabase::getTopics(CompletionHandler<void(PushTopics&&)>&& completionHandler)
 {
     dispatchOnWorkQueue([this, completionHandler = WTFMove(completionHandler)]() mutable {
-        Vector<String> topics;
-        auto sql = cachedStatementOnQueue("SELECT topic FROM Subscriptions"_s);
+        PushTopics topics;
+
+        auto sql = cachedStatementOnQueue(
+            "SELECT sub.topic, ss.state "
+            "FROM Subscriptions sub "
+            "JOIN SubscriptionSets ss "
+            "ON sub.subscriptionSetID = ss.rowid"_s);
         if (!sql) {
             PUSHDB_RELEASE_LOG_BIND_ERROR();
-            completeOnMainQueue(WTFMove(completionHandler), Vector<String> { });
+            completeOnMainQueue(WTFMove(completionHandler), topics);
             return;
         }
 
-        while (sql->step() == SQLITE_ROW)
-            topics.append(sql->columnText(0));
+        while (sql->step() == SQLITE_ROW) {
+            switch (static_cast<SubscriptionSetState>(sql->columnInt(1))) {
+            case SubscriptionSetState::Enabled:
+                topics.enabledTopics.append(sql->columnText(0));
+                break;
+            case SubscriptionSetState::Ignored:
+                topics.ignoredTopics.append(sql->columnText(0));
+                break;
+            }
+        }
 
-        completeOnMainQueue(WTFMove(completionHandler), WTFMove(topics));
+        completeOnMainQueue(WTFMove(completionHandler), topics);
```

Comment:
Why are we dropping the WTFMove here? Doesn't it mean we're losing out of the optimized isolatedCopy() on a r-value?

Reply:
Yeah I accidentally dropped this while refactoring and will add it back in.